cast/js/src/main/resources/prologue.js

```diff
@@ -744,4 +810,82 @@ function EvalError(str) {
 	this.message = new String();
 }
 
+//MAP 
+Map = function Map() {};
+
+local_map.__proto__ = Function.prototype;
+
+Map$proto$__WALA__ = {
+
+  __proto__: Object.prototype,
+
+  constructor: Map,
+
+  has: function Map_prototype_has (key) {
+    // TO DO
+  }
+}
+
+local_map.prototype = Map$proto$__WALA__;
+
+//SET 
+Set = function Set() {};
+
+local_set.__proto__ = Function.prototype;
+
+Set$proto$__WALA__ = {
+
+  __proto__: Object.prototype,
+
+  constructor: Set,
+
+  has: function Set_prototype_has () {
+    // TO DO
+  }
+}
+
+local_set.prototype = Set$proto$__WALA__;
+
+//PROXY 
+Proxy = function Proxy() {};
+
+local_proxy.__proto__ = Function.prototype;
+
+Proxy$proto$__WALA__ = {
+
+  __proto__: Object.prototype,
+
+  constructor: Proxy,
+
+  resolve: function proxy_prototype_resolve () {
+    // TO DO
+  },
+  reject: function Proxy_prototype_reject () {
+    // TO DO
+  },
+  then: function Proxy_prototype_then (onFulfilled, onRejected) {
+    // TO DO
+  },
+}
+
+local_proxy.prototype = Proxy$proto$__WALA__;
+
+Promise = function Promise() {};
+
+local_promise.__proto__ = Function.prototype;
+
+Promise$proto$__WALA__ = {
+
+  __proto__: Object.prototype,
+
+  constructor: Promise,
+
+  then: function Promise_prototype_then (onFulfilled, onRejected) {
+    // TO DO
+  },
+  catch: function Promise_prototype_catch (onRejected) {
+    // TO DO
+  }
+}
 
+local_promise.prototype = Promise$proto$__WALA__;
```

Comment:
Can we add a newline at the end of the file?

Reply:
done